### PR TITLE
Add OCI Artifacts manifest type

### PR DESCRIPTION
**Pull request by aviral26**
This PR adds a new type, `Artifact`, that describes a manifest for an OCI artifact. Spec available here: https://github.com/opencontainers/artifacts/pull/29

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module github.com/notaryproject/artifacts
+
+go 1.15
```
Comment (shizhMSFT) [inline go.mod:3]: The latest is `1.16`

File: specs-go/v2/artifact.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	specs.Versioned
+
+	// MediaType is the media type of the object this schema refers to.
+	MediaType string `json:"mediaType"`
+
+	// ArtifactType is the artifact type of the object this schema refers to.
+	ArtifactType string `json:"artifactType"`
+
+	// Config references the index configuration.
+	Config Descriptor `json:"config"`
```
Comment (shizhMSFT) [inline specs-go/v2/artifact.go:17]: `config` is optional. Should we take `*Descriptor` instead of `Descriptor`?

Comment (shizhMSFT) [reply specs-go/v2/artifact.go:17]: If it is empty, should we do `omitempty`?

File: specs-go/v2/mediatype.go [hunk 3]
```diff
@@ -1,20 +1,6 @@
...
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
 package v2
 
 const (
-	// MediaTypeImageIndex specifies the media type for an image index.
-	MediaTypeImageIndex = "application/vnd.oci.image.index.v2+json"
+	// MediaTypeArtifact specifies the media type for an OCI artifact.
+	MediaTypeArtifact = "application/vnd.oci.artifact.manifest.v1+json"
```
Comment (shizhMSFT) [inline specs-go/v2/mediatype.go:5]: Should it be `MediaTypeArtifact` or `MediaTypeArtifactManifest`?

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (aviral26) [reply go.mod]: Addressed

File: specs-go/v2/artifact.go (hunk 2 above)
Comment (aviral26) [reply specs-go/v2/artifact.go]: Yes, makes sense

File: specs-go/v2/mediatype.go (hunk 3 above)
Comment (aviral26) [reply specs-go/v2/mediatype.go]: Addressed

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Comment by SteveLasker** [comment]
@aviral26, can you move this PR to the [artifacts v1.0-stage branch](https://github.com/opencontainers/artifacts/tree/v1.0-stage)
Once done, we'll remove this fork and track OCI Artifact and subsequent manifest changes there.

**Pull request merged**